connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java

```diff
@@ -129,6 +130,9 @@ public Response putConnectorConfig(final @PathParam("connector") String connecto
                                        final @QueryParam("forward") Boolean forward,
                                        final Map<String, String> connectorConfig) throws Throwable {
         FutureCallback<Herder.Created<ConnectorInfo>> cb = new FutureCallback<>();
+        String includedName = connectorConfig.get(ConnectorConfig.NAME_CONFIG);
+        if (includedName != null && !includedName.equals(connector))
```

Comment:
If the included name is null, then it looks like we depend on the Herder to set it. Would it make sense to move that here?